authorization/invitation/service/invitation_service.go

```diff
@@ -14,24 +20,135 @@ type InvitationService interface {
 }
 
 type InvitationServiceImpl struct {
-	invModelService invitationModel.InvitationModelService
-	db              application.DB
+	InvitationService
+	repo    repository.Repositories
+	permSvc permissionService.PermissionService
 }
 
-func NewInvitationService(invitationModelService invitationModel.InvitationModelService, db application.DB) InvitationService {
-	return &InvitationServiceImpl{invModelService: invitationModelService, db: db}
+func NewInvitationService(repositories repository.Repositories) InvitationService {
+	return &InvitationServiceImpl{repo: repositories, permSvc: repositories.PermissionService()}
 }
 
+// Issue creates new invitations.  The inviteTo parameter is the unique id of the organization, team, security group or resource for
+// which the invitations will be issued, and the invitations parameter contains the users and state for each individual user invitation.
+// This method creates one record in the INVITATION table for each user in the invitations parameter.  Any roles that are issued
+// as part of a user's invitation are created in the INVITATION_ROLE table.
 func (s *InvitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UUID, inviteTo string, invitations []invitation.Invitation) error {
+	var inviteToIdentity *account.Identity
```

Comment:
You removed Model Service (need a better name) and moved its code (as-is) to the Controller Service.
This is not what we agreed to do yesterday. We can remove unnecessary layers but we can't mix them.
Controller Service must use transactions. But this service doesn't use them.
So I see two options here:
1. Move this code back to Model Service (need better name). Remove the Controller Service. Use Model Service directly from Controller. Transaction management should be in the controller then.
2. Wrap the whole thing inside this Issue() method in the Controller Service with a transaction. Remove transaction management from Controller.

IMO the first option is better.